airflow/models.py

```diff
@@ -343,10 +343,15 @@ def kill_zombies(self, session=None):
                 dag = self.dags[ti.dag_id]
                 if ti.task_id in dag.task_ids:
                     task = dag.get_task(ti.task_id)
+
+                    # now set non db backed vars on ti
                     ti.task = task
-                    ti.handle_failure("{} killed as zombie".format(ti))
+                    ti.test_mode = configuration.getboolean('core', 'unit_test_mode')
+
+                    ti.handle_failure("{} detected as zombie".format(ti),
+                                      ti.test_mode, ti.get_template_context())
                     self.logger.info(
-                        'Marked zombie job {} as failed'.format(ti))
+                        'Marked zombie job %s as %s', ti, ti.state)
```

Comment:
@gwax Format provides support for specifying format params both positionally and via named (i.e. keyword) params : http://stackoverflow.com/questions/17895835/format-strings-and-named-arguments-in-python